ui/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.js

```diff
@@ -263,15 +267,58 @@ module.exports = Marionette.LayoutView.extend({
     this.$el.toggleClass('is-maximised', isMaximised(this.goldenLayout.root))
   },
   detectIfGoldenLayoutEmpty() {
-    this.$el.toggleClass(
-      'is-empty',
-      this.goldenLayout.root.contentItems.length === 0
-    )
+    const isEmpty = this.goldenLayout.root.contentItems.length === 0
+    this.$el.toggleClass('is-empty', isEmpty)
+    if (isEmpty) {
+      this.addEmptyVisualizationTab()
+    }
   },
   handleGoldenLayoutInitialised() {
     this.detectIfGoldenLayoutMaximised()
     this.detectIfGoldenLayoutEmpty()
   },
+  handleGoldenLayoutTabCreated() {
+    this.handleTabUpdate(1)
+  },
+  handleGoldenLayoutTabDestroyed() {
+    this.handleTabUpdate(-1)
+  },
+  handleTabUpdate(i) {
+    if (
+      this.goldenLayout.root.contentItems.length > 0 &&
+      this.goldenLayout.root.contentItems[0].header
+    ) {
+      const items = this.goldenLayout.root.contentItems[0].header.tabs.length
+      const width =
+        (this.goldenLayout.container.find('.lm_tabs').width() / items) *
+        (items + i)
+      this.addVisualizationTab(width)
+    }
+  },
+  addVisualizationTab(width) {
```

Comment:
❓ Just curious, why do we need to manually calculate the tab widths and remove / re-add the add visualization button when a tab is updated?

Reply:
Thought this was necessary because of the way golden-layout was render their tabs, but looking at it again I realized I was overcomplicating the issue. Styling has been updating and it is no longer manually doing this